app/services/authorizer.rb

```diff
@@ -25,6 +25,9 @@ def can?(permission, subject = nil, cache = true)
   end
 
   def find_collection(resource_class, options = {})
+    return resource_class.where('1=0') if user.nil?
```

Comment:
I think `find_collection` already knows how to short circuit these special cases in `build_filtered_scope_components`

Reply:
I think, likewise in [authorizable.rb](https://github.com/theforeman/foreman/blob/develop/app/models/concerns/authorizable.rb#L63), I need to move these two lines in helper file.

Reply:
`find_collection` would only be called from `can?` method, which already has a [guard for this case](https://github.com/theforeman/foreman/blob/ef4b3b25f8874b7414616acac473472fc22d0d61/app/services/authorizer.rb#L13-L15)

Reply:
@ares, i this case, should we leave the short circuit, or we can rely on `build_filtered_scope_components` to do the right job?

Reply:
if Authorizer#find_collection will be used, I think I'd add it explicitly here, but after yesterday chat with @kgaikwad, maybe this won't be necessary